framework/kmod.nix

```diff
@@ -1,17 +1,42 @@
-{ config, lib, ... }:
-{
-  options.hardware.framework.enableKmod = lib.mkEnableOption (lib.mdDoc
+{ config, lib, pkgs, ... }:
+let
+  kernel_version_compatible = lib.versionAtLeast config.boot.kernelPackages.kernel.version "6.10";
```

Comment:
One can access config.boot.kernelPackages when setting kernelPatches? I always thought this would result in infinite recursion.

Reply:
it works for me so i guess not. i'm not sure if there's theoretically an issue here, but if this is undefined behavior then i can refactor the patch

Reply:
Maybe patches are applied after the fact.